network/builder.go

```diff
@@ -80,11 +145,11 @@ func (builder *Builder) AddPlugin(plugin PluginInterface) error {
 // misconfiguration, or a *Network.
 func (builder *Builder) Build() (*Network, error) {
 	if builder.keys == nil {
-		return nil, errors.New("cryptography keys not provided to Network; cannot create node ID")
+		return nil, ErrNoKeyPair
 	}
 
 	if len(builder.address) == 0 {
-		return nil, errors.New("Network requires public server IP for peers to connect to")
+		return nil, ErrNoAddress
```

Comment:
If you reused an error, does it store the old stack trace?

Reply:
not sure the interaction with pkg/errors

Reply:
errors.Wrapf() does what you want

Reply:
seems like it doesn't pick it up properly

Reply:
i think it pulls it from the `ErrNoAddress` stacktrace, we probably need the parameters reversed

Reply:
this hack seems to get the stacktrace properly:
```
return errors.Wrapf(ErrNoAddress, "")
```